Shepherds error on uncommitted .loom/ files in main repo

## Problem

All shepherds hit an error on startup because the main repository has untracked files in `.loom/`:

```
[05:20:41] [WARN] Main repository has 2 uncommitted change(s):
[05:20:41] [ERROR] Main repo has uncommitted changes that could cause test
  2. Use --allow-dirty-main to proceed anyway
```

The files triggering this are daemon-generated state files:
- `.loom/daemon-metrics.json`
- `.loom/guide-docs-state.json`
- `.loom/logs/*`

These are runtime artifacts that should be gitignored and not trigger dirty-repo detection.

## Impact

- Every shepherd logs a warning/error on startup
- Adds noise to already-garbled logs
- The `--allow-dirty-main` workaround is applied automatically but shouldn't be needed
- Could cause false failures if the flag isn't set

## Expected Behavior

The dirty-repo check should either:
1. Exclude `.loom/` paths from uncommitted change detection (they're runtime artifacts)
2. Only check tracked files (use `git diff` not `git status` for untracked)
3. Ensure `.loom/daemon-metrics.json`, `.loom/logs/`, etc. are in `.gitignore`

## Observed In

- Loom 0.2.0 (commit c74dff3)
- All 3 shepherds affected

--- a/.gitignore
+++ b/.gitignore
@@ -35,6 +35,7 @@ Thumbs.db
 
 # Loom runtime state (don't commit these)
 .loom-in-use
+.loom-checkpoint
 .loom/.daemon.pid
 .loom/.daemon.log
 .loom/daemon.sock

--- a/loom-tools/src/loom_tools/daemon_v2/actions/shepherds.py
+++ b/loom-tools/src/loom_tools/daemon_v2/actions/shepherds.py
@@ -104,6 +104,7 @@ def _spawn_single_shepherd(
     args = str(issue_num)
     if ctx.config.force_mode:
         args += " --force"
+    args += " --allow-dirty-main"
 
     # Check for existing session and handle it
     if session_exists(shepherd_name):

--- a/loom-tools/src/loom_tools/shepherd/cli.py
+++ b/loom-tools/src/loom_tools/shepherd/cli.py
@@ -171,19 +171,22 @@ EXAMPLES:
 
 
 def _is_loom_runtime(porcelain_line: str) -> bool:
-    """Check if a git status porcelain line refers to a .loom/ runtime file.
+    """Check if a git status porcelain line refers to a loom runtime file.
+
+    Matches both ``.loom/`` directory files and ``.loom-*`` root-level files
+    (e.g. ``.loom-checkpoint``).
 
     Args:
         porcelain_line: A line from ``git status --porcelain``, e.g. ``"?? .loom/daemon-state.json"``
 
     Returns:
-        True if the file path starts with ``.loom/``
+        True if the file path starts with ``.loom/`` or ``.loom-``
     """
     # Format: "XY filename" or "XY filename -> renamed"
     path = porcelain_line[3:] if len(porcelain_line) > 3 else ""
     if " -> " in path:
         path = path.split(" -> ")[-1]  # Use destination for renames
-    return path.startswith(".loom/")
+    return path.startswith(".loom/") or path.startswith(".loom-")
 
 
 def _check_main_repo_clean(repo_root: Path, allow_dirty: bool) -> bool:

--- a/loom-tools/tests/shepherd/test_cli.py
+++ b/loom-tools/tests/shepherd/test_cli.py
@@ -14,6 +14,7 @@ import pytest
 from loom_tools.shepherd.cli import (
     _auto_navigate_out_of_worktree,
     _check_main_repo_clean,
+    _is_loom_runtime,
     _create_config,
     _format_diagnostics_for_comment,
     _format_diagnostics_for_log,
@@ -351,6 +352,37 @@ class TestMain:
             assert result == 0
 
 
+class TestIsLoomRuntime:
+    """Tests for _is_loom_runtime filter (issue #2158)."""
+
+    def test_loom_directory_files(self) -> None:
+        """Should match files under .loom/ directory."""
+        assert _is_loom_runtime("?? .loom/daemon-state.json") is True
+        assert _is_loom_runtime("M  .loom/state.json") is True
+        assert _is_loom_runtime("?? .loom/logs/agent.log") is True
+
+    def test_loom_dash_root_files(self) -> None:
+        """Should match .loom-* root-level files like .loom-checkpoint."""
+        assert _is_loom_runtime("?? .loom-checkpoint") is True
+        assert _is_loom_runtime("M  .loom-in-use") is True
+
+    def test_non_loom_files(self) -> None:
+        """Should NOT match regular source files."""
+        assert _is_loom_runtime("M  src/main.py") is False
+        assert _is_loom_runtime("?? README.md") is False
+        assert _is_loom_runtime("M  .github/workflows/ci.yml") is False
+
+    def test_rename_uses_destination(self) -> None:
+        """Should use destination path for renames."""
+        assert _is_loom_runtime("R  old.txt -> .loom/new.txt") is True
+        assert _is_loom_runtime("R  .loom/old.txt -> src/new.txt") is False
+
+    def test_empty_and_short_lines(self) -> None:
+        """Should handle edge cases gracefully."""
+        assert _is_loom_runtime("") is False
+        assert _is_loom_runtime("M ") is False
+
+
 class TestCheckMainRepoClean:
     """Tests for _check_main_repo_clean pre-flight check (issue #1996)."""
 
@@ -389,6 +421,20 @@ class TestCheckMainRepoClean:
             # Just check it doesn't crash with many files
             result = _check_main_repo_clean(Path("/fake/repo"), allow_dirty=True)
             assert result is True
+
+    def test_filters_loom_checkpoint_file(self) -> None:
+        """Should treat .loom-checkpoint as clean (issue #2158)."""
+        with patch("loom_tools.shepherd.cli.get_uncommitted_files", return_value=["?? .loom-checkpoint"]):
+            result = _check_main_repo_clean(Path("/fake/repo"), allow_dirty=False)
+            assert result is True
+
+    def test_filters_loom_runtime_but_keeps_source_changes(self) -> None:
+        """Should filter .loom/ files but still detect real source changes."""
+        files = ["?? .loom/daemon-state.json", "?? .loom-checkpoint", "M  src/main.py"]
+        with patch("loom_tools.shepherd.cli.get_uncommitted_files", return_value=files), \
+             patch("loom_tools.shepherd.cli.log_warning"):
+            result = _check_main_repo_clean(Path("/fake/repo"), allow_dirty=False)
+            assert result is False
 
 
 def _make_ctx(

--- a/loom-tools/tests/test_loom_shepherd_preflight.py
+++ b/loom-tools/tests/test_loom_shepherd_preflight.py
@@ -293,9 +293,10 @@ class TestIsLoomRuntime:
     def test_rename_out_of_loom(self) -> None:
         assert _is_loom_runtime("R  .loom/old.json -> new.json") is False
 
-    def test_loom_prefix_not_in_dir(self) -> None:
-        """Files named .loom-something (not in .loom/) should NOT be filtered."""
-        assert _is_loom_runtime("?? .loom-config") is False
+    def test_loom_dash_root_files_are_filtered(self) -> None:
+        """Root-level .loom-* files are runtime artifacts and should be filtered."""
+        assert _is_loom_runtime("?? .loom-checkpoint") is True
+        assert _is_loom_runtime("?? .loom-in-use") is True
 
     def test_short_line(self) -> None:
         """Handles pathologically short lines without crashing."""
@@ -349,7 +350,7 @@ class TestCheckMainRepoCleanLoomFiltering:
         with self._mock_uncommitted(["?? .loom/daemon-state.json"]):
             assert _check_main_repo_clean(pathlib.Path("/fake"), allow_dirty=True) is True
 
-    def test_dot_loom_prefix_file_not_filtered(self) -> None:
-        """A file named .loom-config (not in .loom/) should NOT be filtered."""
-        with self._mock_uncommitted(["?? .loom-config"]):
-            assert _check_main_repo_clean(pathlib.Path("/fake"), allow_dirty=False) is False
+    def test_dot_loom_dash_files_are_filtered(self) -> None:
+        """Root-level .loom-* files are runtime artifacts and should be filtered."""
+        with self._mock_uncommitted(["?? .loom-checkpoint"]):
+            assert _check_main_repo_clean(pathlib.Path("/fake"), allow_dirty=False) is True

